src/devices_models/devices/common_models.jl

```diff
@@ -1,3 +1,31 @@
+function DeviceRangeConstraintSpec(
+    ::Type{<:RangeConstraint},
+    ::Type{EnergyShortageVariable},
+    ::Type{T},
+    ::Type{<:AbstractDeviceFormulation},
+    ::Type{<:PM.AbstractPowerModel},
+    feedforward::Union{Nothing, AbstractAffectFeedForward},
+    use_parameters::Bool,
+    use_forecasts::Bool,
+) where {T <: PSY.Component}
+    return DeviceRangeConstraintSpec(;
+        range_constraint_spec = RangeConstraintSpec(;
+            constraint_name = make_constraint_name(
+                RangeConstraint,
+                EnergyShortageVariable,
+                T,
+            ),
+            variable_name = make_variable_name(EnergyShortageVariable, T),
+            limits_func = x -> (
+                min = 0.0,
+                max = PSY.get_energy_shortage_cost(PSY.get_operation_cost(x)) * M_VALUE,
```

Comment:
Why are we multiplying by M here? It feels that this adds to the model constants that the user might not be aware of and can't change. This is delicate and can massively change the behavior of the model.

Reply:
This is to turn off the energy shortage slack when the short cost is zero. This is mainly to avoid unbounded obj func condition when the target energy is zero e.g. in a ED problem where no target is provided but just a value of energy. This bound will never be enforce as the slack is already has ub & lb ( 0 <= slack <= storage capacity), only condition that triggers this is when the slack has no penalty cost.

Reply:
if the shortage cost is zero then zero*M = zero ?

Reply:
yup, this turns off the slack and you can do a energy value formulation. I guess this is a protection again a corner case where the target is zero and shortage slack has no penalty but the surplus slack as a negative cost. So, in this case you have : E + energy_shortage + energy_surplus = E_target => E + energy_shortage = - energy slack . So by taking out energy_shortage slack we are avoiding the unbounded obj func issue

Reply:
Then this should be added in the implementation of the constraint and not here. Can you move the M_VALUE to be used in the implementation of the constraint to catch that corner case and throw a warning?